lib/getstats.js

```diff
@@ -296,25 +300,38 @@ function getTrackStats(peerConnection, track, options) {
 
   if (typeof options.testForChrome !== 'undefined' ||
     typeof webkitRTCPeerConnection !== 'undefined') {
-    return chromeGetTrackStats(peerConnection, track);
+    return chromeOrSafariGetTrackStats(peerConnection, track);
   }
   if (typeof options.testForFirefox  !== 'undefined' ||
     typeof mozRTCPeerConnection !== 'undefined') {
     return firefoxGetTrackStats(peerConnection, track, options.isRemote);
   }
+  if (typeof options.testForSafari  !== 'undefined'
+    || guessBrowser() === 'safari') {
+    if (typeof options.testForSafari  !== 'undefined'
+      || 'currentDirection' in RTCRtpTransceiver.prototype) {
+      return chromeOrSafariGetTrackStats(peerConnection, track);
+    }
+    // NOTE(syerrapragada): getStats() is not supported on
+    // Safari versions where plan-b is the SDP format
+    // due to this bug: https://bugs.webkit.org/show_bug.cgi?id=192601
+    return Promise.reject(new Error(
+      ['getStats() is not supported on this version of Safari',
+       'due to this bug: https://bugs.webkit.org/show_bug.cgi?id=192601'].join(' ')));
```

Comment:
@syerrapragada Nitpick: can you reformat the array `[]` to something like this:
```js
return Promise.reject(new Error([
  'getStats() is not supported on this version of Safari',
  'due to this bug: https://bugs.webkit.org/show_bug.cgi?id=192601'
].join(' ')));
```
It's more readable this way.